### PR TITLE
Enhance discussion of affected[].version formats

### PR DESCRIPTION
**Pull request by andrewpollock**
Based on query raised in https://github.com/google/osv.dev/issues/2135

### CODE REVIEW

File: docs/schema.md [hunk 1]
```diff
@@ -678,6 +678,14 @@ The `affected` object's `versions` field is a JSON array of strings. Each string
 is a single affected version in whatever version syntax is used by the given
 package ecosystem.
 
+When there is no well-defined packaging ecosystem specified, (notably as for
```
Comment (oliverchang) [inline docs/schema.md:681]: Hmm, there is a bit of an assumption here that it's generally accepted/understood that C/C++ libraries will only have GIT commit ranges. Would it make sense to instead say something like:

When there is no well-defined packaging ecosystem specified (for example, general C/C++ libraries), GIT commit ranges typically are the best way to define vulnerable version ranges. In this case, versions specified in this array cannot be relied upon to ...

**Review by oliverchang — COMMENTED** [review]
thanks!

File: docs/schema.md (hunk 1 above)
Comment (andrewpollock) [reply docs/schema.md]: Reworded in b492a29

**Comment by oliverchang** [comment]
@yashrsharma44  tagging you in from https://github.com/google/osv.dev/issues/2135. Would this clarify the question you had in that?

**Comment by yashrsharma44** [comment]
Thanks, this should be helpful :)

**Review by oliverchang — APPROVED** [review]

**Pull request merged**